### PR TITLE
#82 required nullable support

### LINKED ISSUE #82: Support for not required but nullable properties
To use Json encodingDefaults https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/json.md#encoding-defaults feature for PUT requests where we should not send arguments that not change value - we should use default `null` only in case where parameter is `required` and `nullable: true`, or `not required`.

### PR DESCRIPTION
**Pull request by Tetraquark**
Closes #82

### CODE REVIEW

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt [hunk 1]
```diff
@@ -30,7 +31,7 @@ class ProfileApiTest {
     @BeforeTest
     fun setup() {
         json = Json {
-            ignoreUnknownKeys = true
+            encodeDefaults = false
         }
```
Comment (Alex009) [inline sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt:35]: just `json = Json.Default` now will be good :)

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt [hunk 2]
```diff
@@ -140,8 +141,160 @@ class ProfileApiTest {
...
+                id = "10",
+                userSettingsNullableSchema = UserSettingsNullableSchema(
+                    aboutMeRequiredNullable = null,
+                    raitingRequiredNonnull = 10
+                )
+            )
+        }
+
+        assertNotNull(result.stateName)
+        assertTrue(isTestSuccess)
```
Comment (Alex009) [inline sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt:187]: please separate it into multiple tests

Comment (Tetraquark) [reply sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt]: Fixed

File: sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt (hunk 1 above)
Comment (Tetraquark) [reply sample/mpp-library/src/commonTest/kotlin/ProfileApiTest.kt]: Fixed

**Review by Alex009 — APPROVED** [review]

**Review by Alex009 — APPROVED** [review]

**Pull request merged**